ArrayLimitedQueue/README.md

```diff
@@ -0,0 +1,276 @@
+# ArrayLimitedQueue
+
+Array Limited Queue is a collection that has the features of a regular array (only get + limited size), queues and sets. The way you value and compare items can be user-defined.
+
+The Ordered Set data type is a hybrid of:
+- a [Queue](https://en.wikipedia.org/wiki/Queue_(abstract_data_type)), is a list where you can only insert new items at the back and remove items from the front. This ensures that the first item you enqueue is also the first item you dequeue. First come, first serve!
+- a [Set](https://en.wikipedia.org/wiki/Set_%28mathematics%29), a collection of unique items where the order does not matter, and
+- a [Sequence](https://en.wikipedia.org/wiki/Sequence), an ordered list of items where each item may appear more than once.
```

Comment:
A problem that could use `queue` and `set` semantics could make use of this. The thing with data structures is that everything has a use, even if some use cases are more niche than others.  

I took another look at the code and from a performance standpoint, there could be improvements that you can make.

For instance, `maxSize` trims the array by using `removeFirst`. Each time `removeFirst` is called, your array needs to shift all the remaining elements on the left - an **O(n)** operation. (Though if you changed the internal storage type from an array to something else, it won't be called a **Array**LimitedQueue anymore...). A Linked List could be more efficient.

In the `deleteExisting` observer, this is pretty heavy work:

```
internalArray = internalArray.reversed().reduce([]){$0.contains($1) ? $0 : $0 + [$1]}.reversed()
```

- `reversed` is **O(n)**
- `reduce` is **O(n)**
- `contains` is **O(n)**
- `reversed` is **O(n)**

Like I mentioned above, I like the functionality this data structure can provide. The interface seems functionally appealing, but very unswift like at the moment. 

Writing this:

```
maxSize = 10
```

not only defines how many elements this data structure can hold, but also removes anything over that limit from the front of the queue. That makes sense, but feels a bit weird at the same time.